### PR TITLE
Update IAM helper methods to protect against IAM conditions

### PR DESCRIPTION
**Pull request by jskeet**
The methods throw an exception if:

- The policy version is greater than 1
- Any bindings have conditions

The exception message will need to be changed to include a link with more information.

(Note that this is a new PR so I can close #3359 but leave the branch present for now.)

### CODE REVIEW

File: apis/Google.Cloud.Iam.V1/Google.Cloud.Iam.V1/PolicyPartial.cs [hunk 1]
```diff
@@ -77,5 +91,9 @@ public bool RemoveRoleMember(string role, string member)
         }
 
         private Binding FindRole(string role) => Bindings.FirstOrDefault(binding => binding.Role == role);
+
+        private void ValidatePolicyVersion() =>
+            GaxPreconditions.CheckState(Version <= 1 && Bindings.All(b => b.Condition == null),
+                "Helper methods cannot be invoked on policies with version {0} or with conditional bindings. For more information, see FIXME.", Version);
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Iam.V1/Google.Cloud.Iam.V1/PolicyPartial.cs]: Done, thanks.

**Comment by jskeet** [comment]
Amanda, this is now ready to merge - PTAL, and I can give you more context if you'd like.

File: apis/Google.Cloud.Iam.V1/Google.Cloud.Iam.V1/PolicyPartial.cs [hunk 2]
```diff
@@ -26,14 +26,21 @@ public partial class Policy
         /// <summary>
         /// Adds the specified member to the specified role. If the role does
         /// not already exist, it is created.
+        /// This method will fail with an <see cref="InvalidOperationException"/>
+        /// if it is called on a Policy with a <see cref="Version"/> greater than 1,
+        /// or if any of the bindings contain conditions,
+        /// as that indicates a more complicated policy than this method is prepared
+        /// to handle. Changes to such policies must be made manually.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Iam.V1/Google.Cloud.Iam.V1/PolicyPartial.cs]: What does manually mean here, like through the UI?

Comment (jskeet) [reply apis/Google.Cloud.Iam.V1/Google.Cloud.Iam.V1/PolicyPartial.cs]: Discussed offline.

**Pull request merged**